projects/packages/sync/src/modules/class-module.php

```diff
@@ -338,23 +360,31 @@ public function send_full_sync_actions( $config, $status, $send_until ) {
 		$limits = Settings::get_setting( 'full_sync_limits' )[ $this->name() ];
 
 		$chunks_sent = 0;
-		// phpcs:ignore Generic.CodeAnalysis.AssignmentInCondition.FoundInWhileCondition
-		while ( $objects = $this->get_next_chunk( $config, $status, $limits['chunk_size'] ) ) {
-			if ( $chunks_sent++ === $limits['max_chunks'] || microtime( true ) >= $send_until ) {
+
+		$last_item = $this->get_last_item( $config );
+
+		while ( $chunks_sent < $limits['max_chunks'] && microtime( true ) < $send_until ) {
+			$objects = $this->get_next_chunk( $config, $status, $limits['chunk_size'] );
+
+			if ( $wpdb->last_error ) {
 				return $status;
 			}
 
+			if ( empty( $objects ) ) {
+				$status['finished'] = true;
+				return $status;
+			}
 			$result = $this->send_action( 'jetpack_full_sync_' . $this->name(), array( $objects, $status['last_sent'] ) );
 			if ( is_wp_error( $result ) || $wpdb->last_error ) {
 				$status['error'] = true;
 				return $status;
 			}
-			// The $ids are ordered in descending order.
+			// Updated the sent and last_sent status.
 			$status = $this->set_send_full_sync_actions_status( $status, $objects );
-		}
-
-		if ( ! $wpdb->last_error ) {
-			$status['finished'] = true;
+			if ( $last_item === $status['last_sent'] ) {
+				$status['finished'] = true;
```

Comment:
```suggestion
				$status['finished'] = true;
				return $status;
```

Shouldn't we return early here as well?